code/modules/clothing/modular_armor/attachments.dm

```diff
@@ -40,6 +40,9 @@
 	///Light modifier for attachment to an armor piece
 	var/light_mod = 0
 
+	///Replacement for initial icon that allows for the code to work with multiple variants
+	var/base_icon
```

Comment:
Why did you change this to not use inital() anymore?

Reply:
Hyper helped me and we found that the only way to have a swapping module that could use both the Xenonauten sprite and Jaeger sprite was to make a new var. Because initial icon uses the sprite on the code (which is the Jaeger sprite) the parent variant would set it to Xenonauten but activating the welding module would then set it to Jaeger because of how the attachment sprite code worked. Base value could be set to the parent variant which would save the Xenonauten sprite for the sprite changing stuff. Now I probably didn't describe it very well, and I could try showing you over discord if you want.
TLDR: The code that handles sprite change for active and inactive modes needs base icon to save the Xenonauten sprite and not revert to Jaeger like Initial icon does, I can explain it better on discord if you'd like.